tesseract/integration-test/src/lib.rs

```diff
@@ -459,19 +454,40 @@ async fn get_request_works() -> Result<(), anyhow::Error> {
 #[tokio::test(flavor = "multi_thread")]
 #[ignore]
 async fn run_integration_tests() -> Result<(), anyhow::Error> {
+	let _ = log_setup();
 	let (chain_a_sub_client, chain_b_sub_client) = create_clients().await?;
-	let (client_a, _client_b) =
+	let (client_a, client_b) =
 		(chain_a_sub_client.clone().client, chain_b_sub_client.clone().client);
 
+	let mut block_stream = client_a.blocks().subscribe_finalized().await?;
+	while let Some(block_result) = block_stream.next().await {
+		match block_result {
+			Ok(_block) => {
+				log::info!("chain producing blocks and finalizing");
+				break
+			},
+			Err(_err) => {
+				log::info!("chain not yet producing blocks and finalizing");
+				continue
+			},
+		}
+	}
```

Comment:
Just take 1 and end the stream

Reply:
am doing this to make sure I wait until the chain starts to finalize blocks